### PR TITLE
fix(es/transforms/compat): transform optional eval call to indirect eval call

### PR DESCRIPTION
**Pull request by magic-akari**
**Related issue (if exists):**

fix https://github.com/swc-project/swc/issues/2735

### CODE REVIEW

File: crates/swc_ecma_transforms_compat/src/es2020/opt_chaining.rs [hunk 1]
```diff
@@ -501,16 +503,24 @@ impl OptChaining {
                 ..
             }) => {
                 let obj_span = obj.span();
-                let is_super_access = match **obj {
+                let is_super_access = match obj.as_ref() {
```
Comment (kdy1) [inline crates/swc_ecma_transforms_compat/src/es2020/opt_chaining.rs:506]: I think this change is redundant.

File: crates/swc_ecma_transforms_compat/src/es2020/opt_chaining.rs [hunk 2]
```diff
@@ -501,16 +503,24 @@ impl OptChaining {
...
                     Expr::Member(MemberExpr {
                         obj: ExprOrSuper::Super(..),
                         ..
                     }) => true,
                     _ => false,
                 };
 
-                let (left, right, alt) = match **obj {
-                    Expr::Ident(..) => (obj.clone(), obj.clone(), e.expr.take()),
+                let (left, right, alt) = match obj.as_ref() {
```
Comment (kdy1) [inline crates/swc_ecma_transforms_compat/src/es2020/opt_chaining.rs:514]: Same here.

File: crates/swc_ecma_transforms_compat/src/es2020/opt_chaining.rs [hunk 3]
```diff
@@ -501,16 +503,24 @@ impl OptChaining {
...
+                    Expr::Ident(ident) => (
+                        obj.clone(),
+                        obj.clone(),
+                        if *ident.sym == *"eval" {
```
Comment (kdy1) [inline crates/swc_ecma_transforms_compat/src/es2020/opt_chaining.rs:518]: This comparison can be replaced with `ident.sym == js_word!("eval")` which is much faster.

**Review by kdy1 — CHANGES_REQUESTED** [review]
Thank you!
I found some tiny issues.

File: crates/swc_ecma_transforms_compat/src/es2020/opt_chaining.rs (hunk 2 above)
Comment (magic-akari) [reply crates/swc_ecma_transforms_compat/src/es2020/opt_chaining.rs]: I changed it to `&**obj`

**Review by kdy1 — APPROVED** [review]
LGTM. Thank you!

**Pull request merged**